pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/RealtimeSegmentAssignment.java

```diff
@@ -230,7 +249,7 @@ public void init(HelixManager helixManager, TableConfig tableConfig) {
 
       newAssignment = new TreeMap<>();
       for (String segmentName : completedSegmentAssignment.keySet()) {
-        List<String> instancesAssigned = assignSegment(segmentName, consumingInstancePartitions);
+        List<String> instancesAssigned = assignConsumingSegment(segmentName, consumingInstancePartitions);
```

Comment:
Is this a typo? Or are you intentionally calling assignConsumingSegment() method?

Reply:
This is intentional. This code path is for scenarios when completed instance partitions are not provided (no segment relocation), we want to keep the consuming partition on the correct server. This is useful when adding/removing servers for real-time tables.

Reply:
ah got it